1-Draft/RFC00XX-Update-Error-View.md

```diff
@@ -0,0 +1,181 @@
+---
+RFC: Update $ErrorView with Single String Error Messages
+Author: Jason Helmick
+Status: Draft
+Version: 0.0.0
+Area: PowerShell
+Comments Due: 
+---
+
+# Update $ErrorView with Single String Error Messages
+
+When an error occurs in PowerShell, the customers on-screen error message experience currently
+provides a level of detail that obscures the inner exception message from being read by
+new or occasional PowerShell users.
+To improve both comprehension and the troubleshooting experience, a single string of text
+containing the error will be displayed.  A cmdlet will provide the full error details when desired.
+
+## Motivation
+
+The on-screen experience, when receiving an error message,
+is controlled through the views NormalView (the default) and CategoryView. These are user selectable though the preference variable $ErrorView.
+The addition of a “ConciseView” as default that contains only specific error
+information producing a single line on screen and in logs will improve customer success.
+
+Comprehensive error information is available to the customer through
+accessing the error object. For new and occasional customers,
+a cmdlet “Resolve-ErrorRecord” will simplify the process of getting detailed error information, improving customers success.
+
+## Specification
+
+The proposal is to add a new single-line default error message view called ConciseView
+and add a new cmdlet (Resolve-ErrorRecord) to provide detailed error information.
+
+__Key Design Considerations__
+
+1. To reduce confusion and improve debugging success,
+error messages by default should produce a single line of text, including the word “ERROR:” to make consistent with Verbose and Warning messages. This is added to $errorView as view named "ConciseView".
+
+- $ErrorView should contain these views
+
+    + NormalView
```

Comment:
Please remove `View` from these values. It's redundant and doesn't provide value since the user already knows they're working with views from the preference name.

Reply:
I agree that it would be cleaner if the views didn't end with "View" as a suffix; however, if they go that route, internally they would still have to recognize `CategoryView` as a valid name because that is an option (not the default) that users may have set in their profiles, scripts, modules, etc. Otherwise it would be a breaking change.

Reply:
If we were doing this from scratch, I would agree `View` is redundant, but as @KirkMunro noted, this would be a breaking change.  At this point, it makes sense to have it consistent.

Reply:
Well, to be clear, it would only be a breaking change _if you weren't careful about how you implemented it_. i.e. You could keep `CategoryView` around in code for backwards compatibility only, but document/use `Category`, `Default`, `Simple`, etc. as view names going forward. That said, while it's possible, I don't think it's worth it for something that most folks will use so rarely.

Reply:
`$ErrorView` isn't an enum. You can set it to anything, and the only thing that matters is what the format file looks for.

You can trivially match "Category" **and** "CategoryView"

Reply:
Even if we did make it an enum, PowerShell supports partial matching of enum names when casting to those values.

Example from Windows PowerShell:
![image](https://user-images.githubusercontent.com/32407840/62369903-15fbe280-b4ff-11e9-96dd-a9d193ff54ea.png)

Provided we don't add another view name that looks like `'Category'` we can safely allow either usage.

Reply:
Since partial matching works, I would suggest we keep the `View` suffix for consistency and not have both with and without the suffix.

Reply:
If things were consistent we'd have silentlyContinueAction, HighConfirmation and so on.  To be like other preference variables it should not concatenate the preference type and preference value.

Reply:
I also tend to think we should dump `View`. If you look at [PowerShell PR#10435](https://github.com/PowerShell/PowerShell/pull/10435), it's easy to do for legacy handling of `$ErrorView`, but it's nice to be able to drop the redundant text in a completely non-breaking way.

Reply:
This is another one marked as resolved, but is it really resolved? The `View` suffix is redundant. It's a change that can be made without breaking anything, and really, the `$ErrorView` feature wasn't really flushed out before now, so it's highly unlikely that it is used much at all. We have enough changes that we hold back when they are breaking, so why hold back a change that will benefit the community moving forward when it can be made in a non-breaking way? Just because you'd have to maintain backward compatibility on the `CategoryView` name (`NormalView` is just the default/fallback view and that text isn't used in code anywhere)? That's a pretty poor reason to fix this and make it right for now and future versions of PowerShell.